hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/DefaultRecordStore.java

```diff
@@ -720,6 +720,19 @@ private void addEventToQueryCache(Record record) {
         mapEventPublisher.addEventToQueryCache(eventData);
     }
 
+    @Override
+    public void setTTL(Data key, long ttl) {
+        if (mapServiceContext.getNodeEngine().getClusterService().getClusterVersion().isLessThan(Versions.V3_11)) {
```

Comment:
The cluster version check should be as closer to user invocation as possible, at this point a previous-version member would have already thrown a serialization exception due to missing the type ID definition for `SetTTLOperation`. Instead we should check for cluster version:
- in `MapProxyImpl.setTTL` (see also `AbstractDistributedObject.isClusterVersionLessThan` -- these cluster version utility methods should be public so they become useful to proxies)
- in `MapSetTTLMessageTask` (maybe override `beforeProcess()` would be a good place for this check?)

Reply:
done

Reply:
I made `AbstractDistributedObject` class methods protected. I think it makes sense to provide these methods to only implementing classes. However, this class is in `com.hazelcast.spi`. It should be changed cautiously